### ISSUE TITLE
Couldn't find a tree builder with the features you requested: xml

### ISSUE THREAD

**Issue by CydNoxzed** [pelican-plugins/image-process#43]
Hi everyone, 

I tried to use this plugin for pelican and was not able to create the html.

On executing `make html` I got the following error:
```
CRITICAL: Couldn't find a tree builder with the features you requested: xml. Do you need to install a parser library?
Traceback (most recent call last):
  File "~/.virtualenvs/pelican/bin/pelican", line 8, in <module>
    sys.exit(main())
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/__init__.py", line 527, in main
    pelican.run()
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/__init__.py", line 125, in run
    p.generate_output(writer)
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/generators.py", line 685, in generate_output
    self.generate_feeds(writer)
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/generators.py", line 329, in generate_feeds
    writer.write_feed(
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/writers.py", line 165, in write_feed
    signals.feed_written.send(
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/blinker/base.py", line 266, in send
    return [(receiver, receiver(sender, **kwargs))
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/blinker/base.py", line 266, in <listcomp>
    return [(receiver, receiver(sender, **kwargs))
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/pelican/plugins/image_process/image_process.py", line 206, in harvest_feed_images
    soup = BeautifulSoup(f, "xml")
  File "~/.virtualenvs/pelican/lib/python3.9/site-packages/bs4/__init__.py", line 243, in __init__
    raise FeatureNotFound(
bs4.FeatureNotFound: Couldn't find a tree builder with the features you requested: xml. Do you need to install a parser library?
```
I used the version 2.0.0 from [pip](https://pypi.org/project/pelican-image-process/). beautifulsoup4 in version 4.9.3

If I change the Line
https://github.com/pelican-plugins/image-process/blob/8dd305cd080159c88798a7e8e819ba0eb2ab5f2b/pelican/plugins/image_process/image_process.py#L296
from `soup = BeautifulSoup(f, "xml")` to `soup = BeautifulSoup(f, "html.parser")` everything seems to work fine.

Is the xml parser in the function the correct one or should it be the one from the settings?  
```
parser = settings.get("IMAGE_PROCESS_PARSER", "html.parser")
soup = BeautifulSoup(f, parser)
```
If the xml parser is the correct one, shouldn't be there a dependency to [lxml](https://pypi.org/project/lxml/)?

BTW: Thanks for this plugin!

**Comment by kernc** [comment]
This is in:
https://github.com/pelican-plugins/image-process/blob/8dd305cd080159c88798a7e8e819ba0eb2ab5f2b/pelican/plugins/image_process/image_process.py#L282-L296
connected to:
https://github.com/pelican-plugins/image-process/blob/8dd305cd080159c88798a7e8e819ba0eb2ab5f2b/pelican/plugins/image_process/image_process.py#L706
and [RSS/Atom feeds](https://docs.getpelican.com/en/latest/settings.html#feed-settings) _are_ XML.

I wasn't aware bs4 didn't have a simple built-in XML parser. :unamused:

A dependency, or a switch to `html.parser`, or even a simple warning to install lxml. If depend on lxml, we could strip bs4 entirely.

**Comment by MinchinWeb** [comment]
I forget the version, but at some point BeautifulSoup stopped assuming the parser you wanted to use, and made you specify one.

I don't think BeautifulSoup ever carried a build-in parser, relying either on the standard library or an external package like lxml.

**Comment by justinmayer** [comment]
This problem should be fixed in Image Process 2.1.2, which was just automatically published to PyPI via [AutoPub](https://justinmayer.com/projects/autopub/). Many thanks to @MinchinWeb for the fix! ✨
